src/array.rs

```diff
@@ -207,6 +224,47 @@ macro_rules! primitive_array {
                     Ok(vec)
                 }
             }
+
+            impl<This> JavaArrayModificationExt<JavaArray<$rust>, $rust> for This
+            where
+                This: JvmOp,
+                for<'jvm> This::Output<'jvm>: AsJRef<JavaArray<$rust>>,
+            {
+                type SetArrayRegion<'a> = SetArrayRegion<'a, Self, JavaArray<$rust>, $rust>;
+                fn set_array_region<'a>(mut self, start: usize, values: &'a[$rust]) -> Self::SetArrayRegion<'a> {
+                    SetArrayRegion {
+                        this: self,
+                        element: PhantomData,
+                        start,
+                        values
+                    }
+                }
+            }
+
+            impl<This> JvmOp for SetArrayRegion<'_, This, JavaArray<$rust>, $rust>
+            where
+                This: JvmOp,
+                for<'jvm> This::Output<'jvm>: AsJRef<JavaArray<$rust>>,
+            {
+                type Output<'jvm> = ();
+            
+                fn do_jni<'jvm>(self, jvm: &mut Jvm<'jvm>) -> crate::LocalResult<'jvm, Self::Output<'jvm>> {
+                    let this = self.this.do_jni(jvm)?;
+                    let this = this.as_jref()?.as_raw();
+                
+                    unsafe {
```

Comment:
for the unsafe block, please add a safety comment talking about the invariants that are enforced and why this is safe